app/lib/actions/katello/applicability/hosts/bulk_generate.rb

```diff
@@ -2,17 +2,22 @@ module Actions
   module Katello
     module Applicability
       module Hosts
-        class Generate < Actions::EntryAction
+        class BulkGenerate < Actions::EntryAction
           input_format do
             param :host_ids, Array
           end
 
           def run
             input[:host_ids].each do |host_id|
-              ::Katello::EventQueue.push_event(::Katello::Events::GenerateHostApplicability::EVENT_TYPE, host_id)
+              content_facet = ::Host.find(host_id).content_facet
+              content_facet.calculate_and_import_applicability
             end
           end
 
+          def resource_locks
+            :link
+          end
+
```

Comment:
I think we want to add this method to this action:
```
        def queue
          ::Katello::HOST_TASKS_QUEUE
        end
```

so that these run in the 'host queue'